csharp/src/Ice/Connection.cs

```diff
@@ -135,7 +138,7 @@ internal bool Active
             new Dictionary<int, TaskCompletionSource<IncomingResponseFrame>>();
         private Task _sendTask = Task.CompletedTask;
         private State _state; // The current state.
-        private readonly ITransceiver _transceiver;
+        protected readonly ITransceiver Transceiver;
```

Comment:
Resort field/property and prefer a get-only property to non-private field.

It's protected and not `private protected` because you anticipate derived classes outside the Ice assembly?

Reply:
didn't though much about that, but if we allow new transports outside the Ice assembly a new transport might want to create its specialized connection and access this.

Reply:
keep it as protected for now